README.md

```diff
@@ -1,2 +1,213 @@
-Ballerina connector for Microsoft Graph API
+# Module Microsoft Graph
+
+Connects to Microsoft Graph API which acts as the gateway to data and intelligence in Microsoft 365. Microsoft Graph exposes a unified programming model which can be used to access vast amounts of data available in Office 365, Windows 10, and Enterprise Mobility & Security.
+
+The current implementation of Microsoft Graph consists of the following sub modules.
+
+**Spreadsheet Operations**
+
+The `wso2/msspreadsheets` module contains operations to perform CRUD (Create, Read, Update, and Delete) operations on [Excel workbooks](https://docs.microsoft.com/en-us/graph/api/resources/excel?view=graph-rest-1.0) stored in Microsoft OneDrive.
+
+**OneDrive Operations**
+Microsoft OneDrive is a file hosting service and synchronization service run by Microsoft. It is operated as part of Microsoft Office 365.
+The `wso2/msonedrive` module contains operations for accessing the items stored in OneDrive.
+
+## Compatibility
+|                     |    Version     |
+|:-------------------:|:--------------:|
+| Ballerina Language  | 1.2.0   |
+| Microsoftgraph REST API | v1.0          |
+
+## Getting started
+
+1.  Refer the [Getting Started](https://ballerina.io/learn/getting-started/) guide to download and install Ballerina.
+
+2.  To use the Microsoft Graph API, you need to provide the following configuration information in the ballerina.conf file:
+
+       - MS_CLIENT_ID
+       - MS_CLIENT_SECRET
+       - MS_ACCESS_TOKEN
+       - MS_REFRESH_TOKEN
+       - TRUST_STORE_PATH
+       - TRUST_STORE_PASSWORD
+       - WORK_BOOK_NAME
+       - WORK_SHEET_NAME
+       - TABLE_NAME
+
+    Following steps should be followed to obtain the above mentioned configuration information.
+
+    Before you run the following steps you may have to create an account in [OneDrive](https://onedrive.live.com). Next, sign into [Azure Portal - App Registrations](https://portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationsListBlade). You may have to use your personal or work or school account.
+
+    From the App registrations page, click on New registration. Enter a meaningful name in the name field. 
+    In the Supported account types section, select Accounts in any organizational directory and personal Microsoft accounts (e.g. Skype, Xbox, Outlook.com). Click Register to create the application.
+
+    Copy the Application (client) ID (\<MS_CLIENT_ID>). This is the unique identifier for your app.
+    In the application's list of pages (under the Manage tab in left hand side menu), select Authentication.
+    Under the Platform configurations click on "Add a platform" button.
+    Under the "Configure platforms", click on "Web" button located under the Web applications.
+
+    Under the redirect URIs text box, put [OAuth2 Native Client](https://login.microsoftonline.com/common/oauth2/nativeclient).
+    Under the Implicit grant select Access tokens.
+    Click on Configure.
```

Comment:
```suggestion
    Click **Configure**.
```
Shall we add numbered bullets for these steps?

Reply:
Did highlighting the UI element. Since this text is already located within numbered bullets (high-level section numbers) it will be complicated if we use numbered bullets here again.